Lib/argparse.py

```diff
@@ -1124,7 +1128,7 @@ class _VersionAction(Action):
 
     def __init__(self,
                  option_strings,
-                 version=None,
+                 version,
```

Comment:
I think that this change can influence user-defined subclasses in a breaking way.

Reply:
You mean if user builds a class based on `_VersionAction`? The class is not documented and should be an implementation detail, but yes, it the users inherits the class this might break their code.

I guess that's the fundamental question for `argparse` - do we change it at all? Because the realitiy is, if we change it in any kind of way, we might break someone's code, even if it's a bug fix. We can improve the test coverage, but what's right and what's wrong? A wrong test is worse than no test, because the next person trying to fix it would be scared away.

Take https://github.com/python/cpython/issues/103498 as an example, assume we found the missing coverage there, what would we do? Do we write a "coverage test" that honors the current behavior even if it's obviously wrong according to the documentation?

We will find issues when we try to write comprehensive coverage tests, I think that's inevitable. Always keeping the current behavior would make writing tests more difficult (and to be honest, less meaningful).

However, I can revert the code change and just create a new issue for this(I still think this is a bug).